librad/src/git/trailer.rs

```diff
@@ -0,0 +1,273 @@
+// This file is part of radicle-link
+// <https://github.com/radicle-dev/radicle-link>
+//
+// Copyright (C) 2019-2020 The Radicle Team <dev@radicle.xyz>
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License version 3 or
+// later as published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+use std::convert::{TryFrom, TryInto};
+
+#[derive(Debug, Clone, Eq, PartialEq, thiserror::Error)]
+pub enum Error {
+    #[error("The given input is not a well-formed trailer")]
+    IllFormed,
+
+    #[error("Could not parse the trailer due to an invalid token found")]
+    Token(#[from] InvalidToken),
+}
+
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub struct Trailer<'a> {
+    token: Token<'a>,
```

Comment:
Can we make the fields `pub`?

Reply:
I was thinking, instead, to keep them private and provide getters and setters because encapsulation :nerd_face:

Reply:
You already have all the necessary encapsulation by only allowing `Token` to be constructed through a "smart" constructor. More often than not, getters and setters are a PITA in Rust due to "partial borrow" issues and whatnot. This is just a datatype, being able to pattern-match on it is going to be the most convenient way to work with it.

Reply:
@kim, I was joking :hot_face:

Reply:
You! _**shakes fist**_ Next time I'm gonna wrap you in a builder pattern!